server/app/files.go

```diff
@@ -60,7 +60,16 @@ func (a *App) GetFileInfo(filename string) (*mm_model.FileInfo, error) {
 	// filename is in the format 7<some-alphanumeric-string>.<extension>
 	// we want to extract the <some-alphanumeric-string> part of this as this
 	// will be the fileinfo id.
-	fileInfoID := getFileInfoID(strings.Split(filename, ".")[0])
+	parts := strings.Split(filename, ".")
+	if len(parts) < 1 || len(parts[0]) <= 1 {
+		return nil, model.NewErrNotFound("No file found with name" + filename)
+	}
+
+	fileInfoID := getFileInfoID(parts[0])
+
+	if (fileInfoID) == "" {
+		return nil, nil
```

Comment:
Should this case also return an error? I'd expect that this sort of method would always return either a FileInfo or an error, but this case returns nil for both of them

Reply:
Done, returning an error with message `file not found`.